src/test/java/org/broadinstitute/hellbender/tools/copynumber/utils/optimization/PersistenceOptimizerUnitTest.java

```diff
@@ -0,0 +1,87 @@
+package org.broadinstitute.hellbender.tools.copynumber.utils.optimization;
+
+import com.google.common.collect.Ordering;
+import com.google.common.primitives.Doubles;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Random;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * Unit tests for {@link PersistenceOptimizer}.  Recovers local minima of simulated data,
+ * e.g., data produced by adding noise to a sinusoid and a polynomial, along with other simple tests.
+ *
+ * @author Samuel Lee &lt;slee@broadinstitute.org&gt;
+ */
+public final class PersistenceOptimizerUnitTest extends BaseTest {
+    private static final int RANDOM_SEED = 1;   //reset seed before each simulated test case
+
+    @DataProvider(name = "dataPersistenceOptimizer")
+    public Object[][] dataPersistenceOptimizer() {
+        final int numPoints = 100;
+
+        final Random rng = new Random(RANDOM_SEED);
+        final double period = numPoints / 4.;
+        final double[] dataSinusoidal = IntStream.range(0, numPoints).boxed()
+                .mapToDouble(i -> Math.sin(i * 2 * Math.PI / period) + 0.1 * rng.nextGaussian())
+                .toArray();
+        final List<Integer> minimaIndicesExpectedSinusoidal = Arrays.asList(
+                45, 94, 20, 68, 0, 82, 43, 22, 73, 31, 55, 79);                                         //from implementation at https://people.mpi-inf.mpg.de/~weinkauf/notes/persistence1d.html
+
+        rng.setSeed(RANDOM_SEED);
+        final double[] dataPolynomial = IntStream.range(0, numPoints).boxed()
+                .mapToDouble(i -> (double) i / numPoints - 0.5)
+                .map(x -> (x - 0.75) * (x - 0.5) * (x - 0.25) * (x - 0.1) * (x + 0.25) * (x + 0.5) + 0.0001 * rng.nextGaussian())
+                .toArray();
+        System.out.println(Arrays.stream(dataPolynomial).boxed().map(Object::toString).collect(Collectors.joining(", ")));
+        final List<Integer> minimaIndicesExpectedPolynomial = Arrays.asList(
+                8, 68, 99, 73, 82, 89, 36, 34, 79, 45, 38, 49, 43, 84, 60, 56, 66, 62, 70, 92, 47, 87); //from implementation at https://people.mpi-inf.mpg.de/~weinkauf/notes/persistence1d.html
+
+        final double[] dataIdentity = IntStream.range(0, numPoints).boxed()
+                .mapToDouble(i -> i)
+                .toArray();
+        final List<Integer> minimaIndicesExpectedIdentity = Collections.singletonList(0);
+
+        final double[] dataIdentityNegative = IntStream.range(0, numPoints).boxed()
+                .mapToDouble(i -> -i)
+                .toArray();
+        final List<Integer> minimaIndicesExpectedIdentityNegative = Collections.singletonList(numPoints - 1);
+
+        final double[] dataFlat = new double[numPoints];
```

Comment:
Could you add an extra test case, which has a flat part, but also has a single maxima and minima?

Reply:
Good suggestion---the result of that test is actually a little non-intuitive!  Added some extra cases to boot.